Stripe/STPCustomer.m

```diff
@@ -78,13 +78,14 @@ + (nullable instancetype)decodedObjectFromAPIResponse:(nullable NSDictionary *)r
         shipping.name = [shippingDict stp_stringForKey:@"name"];
         shipping.phone = [shippingDict stp_stringForKey:@"phone"];
         NSDictionary *addressDict = [shippingDict stp_dictionaryForKey:@"address"];
-        if (addressDict) {
-            shipping.line1 = [addressDict stp_stringForKey:@"line1"];
-            shipping.line2 = [addressDict stp_stringForKey:@"line2"];
-            shipping.city = [addressDict stp_stringForKey:@"city"];
-            shipping.state = [addressDict stp_stringForKey:@"state"];
-            shipping.postalCode = [addressDict stp_stringForKey:@"postal_code"];
-            shipping.country = [addressDict stp_stringForKey:@"country"];
+        STPAddress *address = [STPAddress decodedObjectFromAPIResponse:addressDict];
+        if (address) {
+            shipping.line1 = address.line1;
+            shipping.line2 = address.line2;
+            shipping.city = address.city;
+            shipping.state = address.state;
+            shipping.postalCode = address.postalCode;
+            shipping.country = address.country;
```

Comment:
I was picturing something more like this:

```objective-c
if (shippingDict) {
    NSDictionary *addressDict = [shippingDict stp_dictionaryForKey:@"address"];
    STPAddress *shipping = [STPAddress decodedObjectFromAPIResponse:addressDict] ?: [STPAddress new];
    shipping.name = [shippingDict stp_stringForKey:@"name"];
    shipping.phone = [shippingDict stp_stringForKey:@"phone"];
    customer.shippingAddress = shipping;
}
```

This version keeps `addressDict` in the `allResponseFields` property, otherwise I think they're the same.

Reply:
oh derp yeah that's better, creating another address was silly